internal/auth0/organization/resource_connection.go

```diff
@@ -26,11 +26,13 @@ func NewConnectionResource() *schema.Resource {
 			"organization_id": {
 				Type:        schema.TypeString,
 				Required:    true,
+				ForceNew:    true,
```

Comment:
What made us include this change within this PR?

Reply:
I noticed we needed it, and didn't want to open a separate PR just for that

Reply:
Let's please open a separate PR. The reason for this is due to us needing to include every behavior change in the changelog and have them be explicit. Someone that comes checking the changelog and sees a reference to this behavior change will have to take a look at a lot of unrelated changes in this PR.